### ISSUE TITLE
[Bug]: Future transactions should not be included in the account balance summary

### ISSUE THREAD

**Issue by tavlima** [actualbudget/actual#2354]
### Verified issue does not already exist?

- [X] I have searched and found no existing issue

### Is this related to GoCardless, Simplefin or another bank-sync provider?

- [ ] I have checked my server logs and could not see any errors there
- [ ] I will be attaching my server logs to this issue
- [ ] I will be attaching my client-side (browser) logs to this issue
- [ ] I understand that this issue will be automatically closed if insufficient information is provided

### What happened?

TL;DR; I'd like to see my balances **as-of today** in the left panel.

I have a lot of future transactions (mostly credit card installments) that I create ahead of time and that is messing up with the accounts balance summary shown in the left panel. I don't think future transactions should impact my **current** balances, as the balances should match what is reported by the bank.

It may be debatable if that is ok for credit card accounts, but given Actual doesn't have any concept of "credit card account", the same rule is applied for everything, which, I think, is wrong.

### Where are you hosting Actual?

None

### Operating System

None

**Comment by youngcw** [comment]
Are you adding those early installments in place of schedules? Schedule are meant for this, were you can see the transactions before hand but they don't actually affect the balances

**Comment by tavlima** [comment]
They are monthly installments, not yearly. And I get schedules are meant
for that, but

1) I couldn’t find any API for that (those are coming from my custom
importer, that pulls from the bank);

2) does it really matter? IMHO, future transactions, no matter the
“source”, should not affect current balance, only forecasts. Indeed, there
is a feature request to treat future transactions just like scheduled.

**Comment by Waseh** [comment]
I agree with this.
I use the gocardless integration to pull transactions from my bank which indeed also adds future transactions (bills scheduled to be paid and so forth) which is messing up the current balance.

**Comment by youngcw** [comment]
I disagree that future transactions should be excluded from the budget.  That would cause lots of confusion where the account shows one thing and the budget shows another.  

I do think it would make sense to not import transactions that are in the future with bank syncing.

**Comment by Teprifer** [comment]
> I disagree that future transactions should be excluded from the budget or account summary. I would expect that to cause more confusion than help. That also would remove the possibility of budget projections.
> 
> I do think it would make sense to not import transactions that are in the future with bank syncing.

I'd seen the comments flow through on this and wasn't sure what I thought for a bit because I could see some reasons each way, but ultimately I'm inclined to agree. If a transaction has been added, then it should affect the balance. Having two different rules for how transactions affect an account doesn't flow.

I can see the view with this being inconsistent with the notion of 'now', but if someone is adding a transaction, it should behave the same as any other transaction to remain consistent in that respect, that it's in the future is beside the point. There are no doubt workflows setup off the current approach for some where they expect their future transactions to affect the balance. so they can 'forecast'. 

Whether these shouldn't be imported at all, or imported as an 'upcoming' record in the same manner of a one off scheduled transactions that can be posted manually (or automatically when the date comes due). I don't know.

**Comment by Kidglove57** [comment]
I agree @Teprifer but I have also noted how my most recent alternative app (Banktivity on Mac) deals with this. It gives the option to display in the sidebar either Today’s Value, Today's Cleared Value  or the Total Value (ie net of all future posted transactions). However, it does include future posted transactions in the budget because they are posted  - but not scheduled transactions.
Apologies if I have misunderstood any of the previous comments but for me, once a transaction is posted on the ledger in Actual, even if with a future date, it will inevitably show both in the budget and in the account balance. Anything else seems to me like a recipe for confusion.

**Comment by tavlima** [comment]
To be clear, I’m not saying those should be excluded from the budget, just
from the accounts totals/summary, as I think it is more intuitive to
display the current values in your bank account, not future. Budgeted
amounts in the future (and scheduled/future transactions for that matter)
seem to work just fine. I’m only talking about the accounts summary.

nYNAB, for example, shows the current balance. GNUCash, as the accounting
app that it is, can display a column with the “posted” and another with the
“current” balance of the accounts.

As a solution, I think a settings/preference option would suffice, as I
don’t think this would change how things are internally calculated too
much, but mostly just displayed. A custom query to the DB with a date
constraint would likely be needed (assuming that is not being computed in
the UI), but not much else.

I hope that clears things up.


Thiago Lima

**Comment by rich-howell** [comment]
I raised this a long time ago as it completely breaks my workflow. Here is an example. 

I get paid on the 23rd of the month, I usually get a pay slip one week before which tells me exactly how much I am going to get paid. 

I enter the amount into the Account Register with a date of the 23rd (which is almost always in the future)

The value that I add appears in my To Be Budgeted and my Account Balance increases by the amount I added. 

This is wrong in my mind, because my bank balance does not yet have that extra money, I have just added it ready.

**Comment by youngcw** [comment]
> To be clear, I’m not saying those should be excluded from the budget, just from the accounts totals/summary, as I think it is more intuitive to display the current values in your bank account, not future. Budgeted amounts in the future (and scheduled/future transactions for that matter) seem to work just fine. I’m only talking about the accounts summary.

How would the be functionally different than looking at the cleared vs uncleared balance?  If the transactions are still included in the budget view then this sounds like the same number as the uncleared balance.

**Comment by rich-howell** [comment]
> > To be clear, I’m not saying those should be excluded from the budget, just from the accounts totals/summary, as I think it is more intuitive to display the current values in your bank account, not future. Budgeted amounts in the future (and scheduled/future transactions for that matter) seem to work just fine. I’m only talking about the accounts summary.
> 
> 
> 
> How would the be functionally different than looking at the cleared vs uncleared balance?  If the transactions are still included in the budget view then this sounds like the same number as the uncleared balance.

Have a look at my example above this reply.

**Comment by youngcw** [comment]
@rich-howell You and Tavlima are giving different examples.  He is saying don't show in the account but do include on budget, you are saying exclude from budget (and maybe account?).

**Comment by Kidglove57** [comment]
Future but unposted regular transactions should (in my view) be dealt with via scheduled transactions - updating the scheduled figure as needed. Isn’t that what schedules are designed for ?

**Comment by Waseh** [comment]
> I disagree that future transactions should be excluded from the budget or account summary. I would expect that to cause more confusion than help. That also would remove the possibility of budget projections.
> 
> I do think it would make sense to not import transactions that are in the future with bank syncing.

From my perspective in regards to the bank syncing just **_not_** importing transactions that are in the future would be my preferred solution, or at least getting the option of not importing transactions that are in the future.

**Comment by youngcw** [comment]
> From my perspective in regards to the bank syncing just **_not_** importing transactions that are in the future would be my preferred solution, or at least getting the option of not importing transactions that are in the future.

@Waseh Could you make a separate feature request or issue ticket for that?  I feel like that should at least be an option.

**Comment by tavlima** [comment]
> How would the be functionally different than looking at the cleared vs uncleared balance

It would be different as I wouldn't have to go through each of the accounts having future transactions to get a hold of their current balance.

---

Let's see how Actual currently behaves... Given this account/transactions...

<img width="1511" alt="image" src="https://github.com/actualbudget/actual/assets/1283363/ceffa41e-a9ea-43ee-89bd-4dd82c297366">

This is what I see in the budget section...

<img width="1508" alt="image" src="https://github.com/actualbudget/actual/assets/1283363/114744bb-3963-4bb7-b807-067f00b537a9">

I honestly can't, for the life of me, understand how some **do not** consider this inconsistent... How can my "To Budget" for February show 100.00, while the account summary itself shows only 75.00?

---

The options, as I see:

## Introduce a settings/preference option on whether to show the posted or current balance in the accounts summary

**Pros**: existing users that are OK with the current behavior can continue with their lives, while users like me (and seemingly others) can switch to more intuitive behavior (as we see it, ofc).

**Cons**: one more setting for the users, which some may consider a problem.


## Introduce an API to create/update scheduled transactions

**Pros**: no behavior change in the app/calculations

**Cons**: switching between two APIs for importing transactions (past vs future) will definitely increase the complexity of the importers.

**Additional notes**: This is good in itself, as exposing additional APIs will definitely be useful (if not to me, but others). In this particular use-case, though, I'm afraid it will be really troublesome to implement, as the API for scheduled transactions should (IMHO) behaves the exact same way as regular transactions in regards to match/deduplication. It is important to have in mind that a transaction that is in the future as of today (and thus a "scheduled transaction") may no longer be tomorrow, when the importer runs again. This will be tricky to handle in the importers as well.

**Comment by youngcw** [comment]
> I honestly can't, for the life of me, understand how some **do not** consider this inconsistent... How can my "To Budget" for February show 100.00, while the account summary itself shows only 75.00?

In this case your budget and account **do have 100 left** in February.  The expenditure only happens in March which shouldn't affect February since that is in the future from February.  By adding those March transactions you are living in the future.  If those transactions weren't there the 100 would be exactly what would be in the account, so if the transactions were added in March, when they posted, the 100 would still be accurate for February and would still be sitting there.

**Comment by rich-howell** [comment]
> Future but unposted regular transactions should (in my view) be dealt with via scheduled transactions - updating the scheduled figure as needed. Isn’t that what schedules are designed for ?
> 
> I have seen other folk say that they want to post all their transactions for the month early to see the effect on their budget and accounts so it sort of feels that the devs can’t win whatever they do. Unless of course an option is given as to which bank balance should be displayed.

You can't make deposits using schedules, at least I can't in my install or in the demo.

**Comment by Teprifer** [comment]
> > You can't make deposits using schedules, at least I can't in my install or in the demo.
> 
> That sounds like a bug. I have my paycheck as a schedule ... hmm looks like this broke again. I fixed that once before. Chrome works for me, but not firefox

Just tested in Chrome desktop, can't toggle the - to a + in the demo.
Tried Chrome mobile(view desktop site) for my own install of three latest release and I can't toggle there either.

**Comment by youngcw** [comment]
> Just tested in Chrome desktop, can't toggle the - to a + in the demo.
> Tried Chrome mobile(view desktop site) for my own install of three latest release and I can't toggle there either.

Ok, so it is only working with existing schedules for me.  So a work around for now is to create a schedule with the wrong amount.  Then edit the schedule and it should work

**Comment by tavlima** [comment]
> So at the end of February your account has $25, TB has 100 and the budget has -75. Which is correct for the month where 100+(-75)=25. Then for March the math is still correct with 125+(-50)=75.

I never questioned the math. From the budget perspective, it looks good enough for me. It is the account summary that is inconsistent, IMO.

**Comment by Kidglove57** [comment]
> > You can't make deposits using schedules, at least I can't in my install or in the demo.
> 
Thanks for spotting this @youngcw. I was puzzled as I have about a dozen scheduled income events.

As an aside I am mulling over how the software should react when used in a way for which it was never designed. I would imagine that this use case would  cause a problem for any envelope budgeting system.

**Comment by youngcw** [comment]
> That sounds like a bug. I have my paycheck as a schedule
> ... hmm looks like this broke again. I fixed that once before. Chrome works for me, but not firefox

For everyone here.  There isn't a bug.  You have to make the amount non-zero before you can change the sign.  Its weird and probably should be changed to not be confusing.

**Comment by tavlima** [comment]
> I would imagine that this use case would cause a problem for any envelope budgeting system.

@Kidglove57, if you are talking about the future transaction issue that is being discussed here, I already called out how that works in other systems.

> nYNAB, for example, shows the current balance. GNUCash, as the accounting app that it is, can display a column with the “posted” and another with the “current” balance of the accounts.

**Comment by tavlima** [comment]
For the reference, this is nYNAB, which I think is much more intuitive and expected.

<img width="1507" alt="image" src="https://github.com/actualbudget/actual/assets/1283363/02997d9e-b9dc-4d7a-9fe2-2e26e9c89d1d">

<img width="1503" alt="image" src="https://github.com/actualbudget/actual/assets/1283363/f99d4d91-242b-41f0-baeb-47d9ce600ce3">

It doesn't make any distinction between future posted transactions or recurrent scheduled expenses: they are all future and (perhaps) subject to change, thus do not impact either budgeted amounts (the expense/income didn't happen yet) or the balances (for the same reason).

**Comment by youngcw** [comment]
@tavlima I would be open to adding something like this as long as its possible to post early, thus applying to the budget.

**Comment by Kidglove57** [comment]
I was referring to @youngcw comments about posting into the following month.

However in fairness to you,  I have never posted transactions more than a day or so in advance - I just look at the prospective running balance alongside the upcoming scheduled transactions to see where I am headed.

I also expand the balance figure at top left of each account register to see what the notional balance on the account is (after all posted transactions) and what the current cleared balance is and what uncleared transactions remain. So I would be referencing the cleared balance as the correct balance on my bank account.

I have no objection to a menu giving users the choice as to which balance is displayed in the sidebar. Although I do want to retain as a default the basic logic that:
Balance of all  "for budget" accounts =
Balance available in budget + income held for next month

**Comment by tavlima** [comment]
> @tavlima I would be open to adding something like this as long as its possible to post early, thus applying to the budget.

Couldn't that also be gated by the same settings?
![image](https://github.com/actualbudget/actual/assets/1283363/1c421828-3b2c-4c90-99d4-63de95e95b83)

**Comment by youngcw** [comment]
yes it could.  Im just saying that the option needs to be there before any changes are made

**Comment by Kidglove57** [comment]
> It doesn't make any distinction between future posted transactions or recurrent scheduled expenses: they are all future and (perhaps) subject to change, thus do not impact either budgeted amounts (the expense/income didn't happen yet) or the balances (for the same reason).

I recall this now from nYNAB (its a while ago for me) - i always explained it to myself as nYNAB treating any future dated transaction as if it were a one-off scheduled transaction.

I looked at another envelope budgeting app (Moneywell on Mac). That displays in a way that would suit your use case - it treats all future dated transactions as pending and not part of the sidebar balance. Hands up, I have used Actual since April 2019 and got used to its way of doing things and as a result rarely post future dated transactions. As I say, I just use the arrows to expand the account balance view at the top of each account register so as to easily see what the cleared and uncleared balances are.

**Comment by rich-howell** [comment]
The moment this becomes a feature request it will loose any traction it has and just fall into the endless pit of ideas. Hopefully someone is willing to pick this up before that happens. 

nYNAB does exactly what i would like, if I create a one of schedule for a future posted transaction, after the posted date will it still appear in the schedules list? So I could end up with hundreds of schedules in a list that potentially have passed?

**Comment by Kidglove57** [comment]
> So I could end up with hundreds of schedules in a list that potentially have passed?

My experience is that a completed schedule is hidden from the schedules list - but I take your point. I only need a very few  one off future dated transactions a year so I guess this is a case of various users commenting who have very different use cases. In fairness to the original Dev I assume he felt that the display (by clicking the arrows) of the cleared, uncleared balances at the top of each register would address this?

 HOWEVER I am all for pending transactions being marked as such and not included in the account balance OR a choice being given to the user. 

A question - would this not mean that the available balances in the budget (net of future transactions) would be out of line with the recorded balance for the “for budget” accounts which could lead to difficulties for users reconciling this. Unless a total posted balance is also displayed somewhere?

**Comment by Waseh** [comment]
> > From my perspective in regards to the bank syncing just **_not_** importing transactions that are in the future would be my preferred solution, or at least getting the option of not importing transactions that are in the future.
> 
> @Waseh Could you make a separate feature request or issue ticket for that? I feel like that should at least be an option.

Sure thing, just did - #2361

**Comment by xentara1** [comment]
Stupid question and apologies if already covered. 
should this not be handled by cleared vs uncleared transactions. 
i.e future transactions are not cleared. 
This can then be seen in the uncleared breakdown. (Setting can be set to adjust if cleared or uncleared is default view)
Then once the transaction actually clears it can be marked as such and thus feed the cleared balances. 

This way the only change needs to be a setting adjusting if the default view is for cleared balances only or all balances.

**Comment by Kidglove57** [comment]
> This way the only change needs to be a setting adjusting if the default view is for cleared balances only or all balances.

That is certainly the way I see it too.

**Comment by Teprifer** [comment]
> > This way the only change needs to be a setting adjusting if the default view is for cleared balances only or all balances.
> 
> That is certainly the way I see it too.

Can already add a filter, but it doesn't save.

I'm not sure repurposing the cleared flag is ideal. It would make it difficult for anyone with future transactions, and current transactions that haven't bank cleared yet.

Any proposal, in my view, shouldn't break existing workflows and app behaviour.

I think the ideal method is some form along the lines of how scheduled transactions show before they're due. Similar to YNAB upcoming.

**Comment by tiagodj** [comment]
Here are my 2 cents. This is a practical example of mine, that brought me here to this thread.

I have multiple checking accounts. I have bills and credits in both of them. The option of moving everything to a single account, for me, is not ideal.

So, a scenario that I see myself often is: how much money do I need in this account in the foreseeable future, in order to cover incoming chargers?

For schedules (expected and known value transactions), it works fine. I can see the Upcoming transactions.

But for credit card payments, it is trickier. CC payments are not really payments, are transfer. So the money for the purchases made in the card are already accounted for in the budget, but are not necessarily in the correct account where the transfer payment will come out of.

Now, I've received the CC bill and I know it will be debited, for example, in 2 and a half weeks. But I don't have the money in the right account right now, for example because I am paid bi-weekly and haven't received the money yet.

So in this scenario, I'd like to enter a **future** transfer transaction, with the CC bill amount, so that I can see in the upcoming list that I will need to have this money ready by that day.

If I already have enough money to move beforehand, sure, I could do that. But sometimes I don't have it right away, although there is a pay check coming in before that transaction date, which gives me a **heads up** to transfer the money in time. 

And this is the scenario where seeing the future transaction is handy (for me).

An extra bonus feature that could spawn from this is to have some kind of Alert in the UI showing that I have a transaction in X days for a specific account, but there are insufficient funds in that account (is there such a functionality? I haven't been in this scenario yet to see it).

Being able to see the future is very useful, and I believe it should be easier to manage that. I understand that the envelop system uses the money we have right now, and I agree with it. But that doesn't stop us from adding this extra layer to prepare for what's coming up.

**Comment by cati-chatou** [comment]
Another case of future transactions is as following: 

I know that I will have an expense on specific date, I know the amount. It is recurring but not regular expense, it does not make sense to put it on schedule. I want to see it in my future expenses so I can plan additional budget money for it. But I don't want it to impact my accounts balances or budget until the transaction actually happen, especially that in might be more than month in future. 

Toolkit for YNAB additionally shows up sum of future transactions in specific month for category. This is very useful for budgeting flow: planning budget - using template as usual - seeing that I have this additional expense planned - allocating additional money.

The way Actual does it currently is plenty confusing - I have a planned (future) transaction and it treats it as "Spent", affecting budget and accounts (I caught myself few times thinking: oh no, I didn't budgeted for this - while I did, it just already have "eaten" that budget, treating it as already spent). 

I think that part of budgeting is to plan for the future - be it saving, a transfer between accounts to assure that I will have the payments covered, adjusting budget to planned transactions.

**Comment by Elarcis** [comment]
Hi, chiming in with a personal use case:

In my country, mortgages payments are scheduled when I contract my loan — I know exactly what I’m going to pay each month for the next 10 years!

But the amount varies for each payment — each month it’s a little lower due to fees being higher at the start of the loan. I also have an off-budget account representing my loan, with a negative balance — excluding bank fees. In YNAB I just scheduled my transactions for the whole year as a split transaction, a set amount of which goes into my “loan” account, the rest goes to my bank. This represents more than € 7,000 each year, which makes a non-negligeable impact on my account balance on Actual.

Doing so with schedules and rules is _incredibly tedious_. I need to create a schedule for each month with a precise amount, then edit it as a rule and use the rule editing UI to set that to a split transaction.

Some features ideas that would help in these cases:

- Allow duplicating schedules and/or rules (especially one-off schedules)
- Allow inputing split transactions in schedules (or even just transactions, with note and category, automatically translated into a rule)
- Facilitating split transactions in rules, like showing the split transaction editor (with payee, category and notes) with the transaction total amount locked, for example.

**Comment by jfdoming** [comment]
> Hi, chiming in with a personal use case:
> 
> In my country, mortgages payments are scheduled when I contract my loan — I know exactly what I’m going to pay each month for the next 10 years!
> 
> But the amount varies for each payment — each month it’s a little lower due to fees being higher at the start of the loan. I also have an off-budget account representing my loan, with a negative balance — excluding bank fees. In YNAB I just scheduled my transactions for the whole year as a split transaction, a set amount of which goes into my “loan” account, the rest goes to my bank. This represents more than €7,000 each year, which makes a non-negligible impact on my account balance on Actual.
> 
> Doing so with schedules and rules is _incredibly tedious_. I need to create a schedule for each month with a precise amount, then edit it as a rule and use the rule editing UI to set that to a split transaction.
> 
> Some features ideas that would help in these cases:
> 
> * Allow duplicating schedules and/or rules (especially one-off schedules)
> * Allow inputting split transactions in schedules (or even just transactions, with note and category, automatically translated into a rule)
> * Facilitating split transactions in rules, like showing the split transaction editor (with payee, category and notes) with the transaction total amount locked, for example.

I see value in the use case for sure! That said, as of a few releases ago you can input splits in rules (and also in schedules by clicking "Edit as rule"). Have you had a chance to try that? Is something not working as it should? (Disclosure: I'm the person who worked on this feature, so I'm interested in feedback 😅)

**Comment by omnizach** [comment]
I would like to import future transactions that align with loan payments. The issue here is that the interest/principal amounts change each payment, so it's not possible to schedule them. I get that the actual payment is constant, but that's more of the on-budget use case. For the actual mortgage and tracking the loan balance (an off-budget account), you need this break down or the balance isn't reflected accurately.

**Comment by mullermn** [comment]
I want to add a vote to the general sentiment that Actual should handle future dated transactions rather than ignoring them as has been suggested in some comments above. I know this can be a contentious topic among budgeters and I won't rehash my personal reasoning for this here, but I just want to add a tick in the 'for' column.

**Comment by mullermn** [comment]
Apologies for replying to myself, this is an independent issue that has just come up that relates to this - I think Actual needs to review how reconciliation works with respect to future dated transactions. Currently, I can't pick the point in time against which to reconcile, but since Actual is aware of transactions that have-but-haven't happened yet, this creates a rather dumb situation where I have to pounce at the right moment when Actual's view of the balance coincides with the bank.

Right now I would like to reconcile but I can't because I'm out to the tune of £4, so Actual wants to create a reconciliation transaction - which would appear right next to the genuine, uncleared transaction that Actual is already aware of!

**Comment by matt-fidd** [comment]
> Apologies for replying to myself, this is an independent issue that has just come up that relates to this - I think Actual needs to review how reconciliation works with respect to future dated transactions. Currently, I can't pick the point in time against which to reconcile, but since Actual is aware of transactions that have-but-haven't happened yet, this creates a rather dumb situation where I have to pounce at the right moment when Actual's view of the balance coincides with the bank.
> 
> Right now I would like to reconcile but I can't because I'm out to the tune of £4, so Actual wants to create a reconciliation transaction - which would appear right next to the genuine, uncleared transaction that Actual is already aware of!
> 
> ![Image](https://github.com/user-attachments/assets/2070330e-bc24-4761-b437-268dc512e85d)

Future transactions should be handled with schedules, not transactions dated into the future.
If that transaction has posted at your bank, then you can mark it as cleared in Actual to match and reconciliation will work as expected.
If that transaction has not yet posted at your bank then the cleared balance is correct at 8003.48 and you are inputting the cleared+pending balance, which will not work.

**Comment by mullermn** [comment]
> Future transactions should be handled with schedules, not transactions dated into the future. If that transaction has posted at your bank, then you can mark it as cleared in Actual to match and reconciliation will work as expected. If that transaction has not yet posted at your bank then the cleared balance is correct at 8003.48 and you are inputting the cleared+pending balance, which will not work.

OK, I was assuming that tampering with the bank-sync imported data was a bad idea, but if that's an acceptable workflow that does work around this. Thanks

**Comment by SalocinHB** [comment]
> OK, I was assuming that tampering with the bank-sync imported data was a bad idea, but if that's an acceptable workflow that does work around this. Thanks

With v2025.3 you can disable importing uncleared transactions. Then those transactions won’t show up until they've actually cleared in your bank account.

**Comment by felipewnp** [comment]
+1 on this.

Doesn't make sense to just "ignore" something that is clearly not right.

If it's not yet in the "envelope" it shouldn't show as it is.

**Comment by AndyDaSilva52** [comment]
@cati-chatou said something that is important to this discussion

> Toolkit for YNAB additionally shows up sum of future transactions in specific month for category. This is very useful for budgeting flow: planning budget - using template as usual - seeing that I have this additional expense planned - allocating additional money.

This is true, even without Toolkit for YNAB (extension) YNAB shows upcoming transactions (schedules) to help budget [Month Ahead Strategy](https://actualbudget.com/docs/getting-started/envelope-budgeting#the-month-ahead-strategy).

![Image](https://github.com/user-attachments/assets/932d0c95-14d3-4d3e-9c5a-1bf23c9cc72d)

---

Considering the layout of Actual that allows to budget 2 or more months at the same time, the Month Ahead Strategy has a much better experience than other apps.

But we need to see how much a category needs in the next month.

Currently, with **future transactions** it has a benefit to see the upcoming transactions and try the [Month Ahead Strategy](https://actualbudget.com/docs/getting-started/envelope-budgeting#the-month-ahead-strategy) to cover the spend that is already an obligation because it's money that was borrowed, so I'm [carrying debt](https://actualbudget.com/docs/budgeting/credit-cards/carrying-debt) and have to deal with that in the upcoming month.

![Image](https://github.com/user-attachments/assets/24557aa2-eae2-4b27-8b9c-6b68b4e49439)

But the **Account Balance** calculation doesn't consider **Cleared/Reconciled** or **Date** (until today), even though is debatable in the case of [**On Budget Account for Credit Card**](https://actualbudget.com/docs/accounts/#:~:text=On%20budget%20accounts%20affect%20the%20budget%2C%20and%20transactions%20can%20be%20categorized.%20These%20are%20accounts%20where%20you%20want%20to%20track%20cash%20flow%20and%20use%20the%20budget%2C%20like%20checking%20accounts%20and%20credit%20cards.) because some will want to see all the money that was borrowed (all future debt that will be needed to be paid in the coming months), and others will want to see what was **Cleared/Reconciled**.

It makes me think if an **Off Budget Account** would offer a different way to manage the credit card like was asked on #1995

---

About **Schedules**
It is supposed to be the feature to deal with **future transactions** with or without a limit of repetition/date (much better than YNAB by the way).

But the experience overall is not align with the [Month Ahead Strategy](https://actualbudget.com/docs/getting-started/envelope-budgeting#the-month-ahead-strategy), see #4815 for consideration.

![Image](https://github.com/user-attachments/assets/57b0c42b-e25a-4820-925e-de47f98a0411)

⚠️ I had to [Edit as Rule the schedule to define the category](https://actualbudget.com/docs/schedules#how-to-use-rules-with-schedules:~:text=By%20clicking%20the%20%2B%20arrow%20on%20in%20the%20Then%20apply%20these%20actions%20area%2C%20you%20can%20define%20the%20category%20this%20schedule%20should%20be%20assigned%20against%20as%20well%20as%20any%20notes%20you%20might%20want%20to%20include.) even though I used an existing transaction that already has the category defined to create the schedule (some improvement for the future).

![Image](https://github.com/user-attachments/assets/b9c5af36-9011-4d81-af21-5481bf7e4c7e)

---

So, I think it will be a step backwards if somehow prevents transactions from being inserted in the future, forcing to being schedules. By the way, for comparison, as far I remember, YNAB doesn't import future transactions and this is one of the reasons I struggle to keep using it because the schedules are very challenging to maintain in their app.

💡 From my point of view, It would be nice to consider only showing the account balance for **Cleared/Reconciled** transactions, it makes sense, and it will work very well, specially with **On Budget Account**.

**Comment by kabo** [comment]
Would it be very hard to have a setting, "only include cleared transactions in the sidebar balance" or similar?

**Comment by alex-t-grey** [comment]
> Would it be very hard to have a setting, "only include cleared transactions in the sidebar balance" or similar?

You can already filter transactions by Cleared status, and save that for reuse. Or do you mean in the budget view? I don't think that's a good idea. In theory, you entered the transaction because you know that it's real, so it's important to have that reflected in your budget. You know that you spent the money, even if it takes a few days for the bank to acknowledge that it was processed.

**Comment by cati-chatou** [comment]
> > Would it be very hard to have a setting, "only include cleared transactions in the sidebar balance" or similar?
> 
> You can already filter transactions by Cleared status, and save that for reuse. Or do you mean in the budget view? I don't think that's a good idea. In theory, you entered the transaction because you know that it's real, so it's important to have that reflected in your budget. You know that you spent the money, even if it takes a few days for the bank to acknowledge that it was processed.

I think it was about account balances side bar view. And it is indeed a simple solution for those who do not want to see future transactions accounted in today’s balance. I for one. I enter the transaction because I know it would happen, be it tomorrow or in week or month time. For example irregular doctor’s appointment - it doesn’t make sense to put it on schedule, I know the amount required already, I put it there to not forget to budget for it, but I would much prefer for it not to skew my today’s balances.

**Comment by alex-t-grey** [comment]
@cati-chatou scheduled transactions aren't strictly for recuring transactions, they're basically just reminders for known future transactions. Your example definitely one of the use cases for scheduled transactions.

That said, I definitely understand the friction of needing to go into schedules for that instead of simply entering the transaction like normal. One thing I really like about nYNAB's implementation is that you created future/scheduled transactions the same way as any other. If you make a transaction with a future date, it just become a "scheduled transaction" automatically. You get a warning on the budget screen that there's an upcoming transaction, but it does not affect your account balances or appear as "spent" unless you choose to post the transaction early.

I would love it if Actual could implement a similar seamless workflow. You'd still have the regular schedules interface when you want that extra granular control.

**Comment by mullermn** [comment]
I don't think this is a question with one 'correct' answer. If I'm reviewing how my finances are today, I don't want transactions that haven't occurred yet to factor in to those numbers because that future might yet change, and because Actual's numbers wouldn't reconcile with any other party like my bank.

However, if I'm doing something forward-facing, like budgeting (which is after all the entire reason software like Actual exists), I absolutely do want those transactions that I 'know' are going to happen to factor in those numbers, because otherwise I'm just doing mental busy work to remember that the 100.00 I have in that category is really 80 because I'm going to spend 20 next week.

I think you need a top level toggle that turns on/off whether future dated transactions are factored in, and that toggle would affect both account balances and category balances in the budget view. 'Show me how things are now' vs. 'show me how things are if they play out as I think they will'.

**Comment by ngist** [comment]
> > Hi, chiming in with a personal use case:
> > In my country, mortgages payments are scheduled when I contract my loan — I know exactly what I’m going to pay each month for the next 10 years!
> > But the amount varies for each payment — each month it’s a little lower due to fees being higher at the start of the loan. I also have an off-budget account representing my loan, with a negative balance — excluding bank fees. In YNAB I just scheduled my transactions for the whole year as a split transaction, a set amount of which goes into my “loan” account, the rest goes to my bank. This represents more than €7,000 each year, which makes a non-negligible impact on my account balance on Actual.
> > Doing so with schedules and rules is _incredibly tedious_. I need to create a schedule for each month with a precise amount, then edit it as a rule and use the rule editing UI to set that to a split transaction.
> > Some features ideas that would help in these cases:
> > 
> > * Allow duplicating schedules and/or rules (especially one-off schedules)
> > * Allow inputting split transactions in schedules (or even just transactions, with note and category, automatically translated into a rule)
> > * Facilitating split transactions in rules, like showing the split transaction editor (with payee, category and notes) with the transaction total amount locked, for example.
> 
> I see value in the use case for sure! That said, as of a few releases ago you can input splits in rules (and also in schedules by clicking "Edit as rule"). Have you had a chance to try that? Is something not working as it should? (Disclosure: I'm the person who worked on this feature, so I'm interested in feedback 😅)

@jfdoming 

I just started with actual a few days ago and discovered the ability split transactions in rules, it worked beautifully for my mortgage use case. I setup one account for the loan and another for escrow balance. The mortgage payment is fixed but escrow changes every now and again. I was able to split up the last year of transactions automatically and easily. 

The issue i'm struggling with now is how to best subdivide the principle and interest since they change every month, I looked at actual-helpers project but it won't backfill data. I can easily generate a spreadsheet of transactions and import for the historical data, and possible future transactions, but I'm worried that will cause me issues with correct balance computation. Maybe I'll try using actual-helpers after I manually backfill.

**Comment by ksmithbaylor** [comment]
Throwing my hat in the ring on the "future transactions" issue, I like to enter my expected income a few months ahead of time so I can budget out a few months. Schedules aren't reflected as income in future months' budgets, so what I do is have an on-budget "Future income" account. When those paychecks (or whatever) actually hit my accounts, I move them from that account to whichever account they should be in. This lets me budget ahead but still keep each account accurate to what's there already.

For me, it would be useful to be able to select subsets of accounts to display totals for, or at least exclude specific accounts from the total.

**Comment by rtbick** [comment]
I enter all transactions manually, so each of these things is really important to my workflow:

- Confirming the account's total in Actual matches the real account, as a way to check for errors
- Scheduling transactions as much as possible (both one-time and recurring)
- Using the mobile app to enter a transaction soon after it occurs, instead of waiting for access to my PC

The mobile app currently doesn't allow access to Schedules [(see my feature request here)](https://github.com/actualbudget/actual/issues/5938#issue-3522787749). So, if I need to add a future transaction on mobile, I must add it as a "normal" transaction. This makes the account balance no longer match the real account, and adds cognitive overhead for the next time I look at my budget on PC, where I need to set it up properly as a Schedule.

> One thing I really like about nYNAB's implementation is that you created future/scheduled transactions the same way as any other. If you make a transaction with a future date, it just become a "scheduled transaction" automatically. You get a warning on the budget screen that there's an upcoming transaction, but it does not affect your account balances or appear as "spent" unless you choose to post the transaction early.
> 
> I would love it if Actual could implement a similar seamless workflow. You'd still have the regular schedules interface when you want that extra granular control.

I agree strongly with @atgrey24 here, as a user coming from nYNAB. I find it very intuitive for a transaction with a future date to be automatically be considered a scheduled transaction. In fact, I think that entering future transactions as way to project account balances contradicts the envelope budgeting philosophy. I'm accustomed to only budgeting the funds that I currently have. 

I also feel that the Transaction workflow to add a one-time future transaction is more intuitive than going into the Schedules page. The Schedule options, while powerful, feel like overkill for one-time transactions. 

Would auto-converting future-dated transactions into Scheduled transactions solve @tavlima's use case? Or should I make a separate issue to discuss that idea?

**Comment by Crosis47** [comment]
I feel an easy quick fix for this would be to just add a "year-to-date" balance that shows the running balance as of today's date in the balance information that is available when clicking on the balance under the account name.

I have just begun the process of switching from Quicken to Actual and I am currently doing this by filtering transactions to only include less than or equal to today and then the filtered balance is my year to date but this should honestly just be always visible as having to do these steps to just see how much money I actually have right now is kind of silly.

**Comment by youngcw** [comment]
A new feature was just added to Actual that allows a user to create a non-repeating schedule directly from the account view.  The option will show up anytime a transaction has a date in the future.  This way a user can decide if they want the upcoming transaction to be a regular transaction, and included in the account balance and budget, or the transaction to become a schedule and not be included in the account balance or budget.  If the transaction is turned into a schedule then it will show up in the account page with other upcoming schedules.

The feature is currently available on the edge build.  You can try it using https://edge.actualbudget.org

**Comment by kabo** [comment]
Hmm, that new feature doesn't help me at least.
I've started doing the workaround described by ksmithbaylor here
https://github.com/actualbudget/actual/issues/2354#issuecomment-3217412459

A bit clunky but gets the job done. It would be nice if there was just a setting for the sidebar totals to only include cleared transactions...

**Comment by SLahti** [comment]
> I enter all transactions manually, so each of these things is really important to my workflow:
> 
> * Confirming the account's total in Actual matches the real account, as a way to check for errors
> * Scheduling transactions as much as possible (both one-time and recurring)
> * Using the mobile app to enter a transaction soon after it occurs, instead of waiting for access to my PC
> 
> The mobile app currently doesn't allow access to Schedules [(see my feature request here)](https://github.com/actualbudget/actual/issues/5938#issue-3522787749). So, if I need to add a future transaction on mobile, I must add it as a "normal" transaction. This makes the account balance no longer match the real account, and adds cognitive overhead for the next time I look at my budget on PC, where I need to set it up properly as a Schedule.
> 
> > One thing I really like about nYNAB's implementation is that you created future/scheduled transactions the same way as any other. If you make a transaction with a future date, it just become a "scheduled transaction" automatically. You get a warning on the budget screen that there's an upcoming transaction, but it does not affect your account balances or appear as "spent" unless you choose to post the transaction early.
> > I would love it if Actual could implement a similar seamless workflow. You'd still have the regular schedules interface when you want that extra granular control.
> 
> I agree strongly with [@atgrey24](https://github.com/atgrey24) here, as a user coming from nYNAB. I find it very intuitive for a transaction with a future date to be automatically be considered a scheduled transaction. In fact, I think that entering future transactions as way to project account balances contradicts the envelope budgeting philosophy. I'm accustomed to only budgeting the funds that I currently have.
> 
> I also feel that the Transaction workflow to add a one-time future transaction is more intuitive than going into the Schedules page. The Schedule options, while powerful, feel like overkill for one-time transactions.
> 
> Would auto-converting future-dated transactions into Scheduled transactions solve [@tavlima](https://github.com/tavlima)'s use case? Or should I make a separate issue to discuss that idea?

I agree with all of this. I also use Actual in a similar fashion.

> A new feature was just added to Actual that allows a user to create a non-repeating schedule directly from the account view. The option will show up anytime a transaction has a date in the future. This way a user can decide if they want the upcoming transaction to be a regular transaction, and included in the account balance and budget, or the transaction to become a schedule and not be included in the account balance or budget. If the transaction is turned into a schedule then it will show up in the account page with other upcoming schedules.
> 
> The feature is currently available on the edge build. You can try it using https://edge.actualbudget.org

Isn't that a bit flawed? Shouldn't schedules be reflected in the budget? That way you know what you need to allocate to cover the expected spending(isn't that the purpose of schedules?). Maybe it could be decided by the "Upcoming length" setting or just show none/all schedules in the budget. Otherwise the purpose of schedules seems to me as just a list of things you need to remember to budget for. My list of schedules is very long so this is a time consuming effort. Sure I can use upcoming length to see that my account balances can cover the upcoming spendings but it doesn't help me with budgeting/planning. Sorry about the rant, I do love Actual despite this issue.

**Comment by Juulz** [comment]
I'd be happy just to get a darker line (red, blue, purple, dark gray) in the transaction lists demarcating below anything newer than today. I can use the running balance to see today's balance.

**Comment by Crosis47** [comment]
> I'd be happy just to get a darker line (red, blue, purple, dark gray) in the transaction lists demarcating below anything newer than today. I can use the running balance to see today's balance.

I made this exact request and was told that's what schedules are for.

**Comment by cati-chatou** [comment]
I sort of gave up.

So now I'm using schedules for some recurring-regular-always-same-amount expenses, but... for all the other expected transactions (including regular but amount changing payments) I'm using either external notes (pity) or I'm adding them to templates. 

I still think that expected transactions impacting current balance is just wrong. But Actual budget is still useful, robust and free, so I can live with that.

**Comment by d2718nis** [comment]
Here's how YNAB does it:
<img width="1415" height="879" alt="Image" src="https://github.com/user-attachments/assets/8b2b5d84-d6d5-47d6-9879-6d287c60b408" />

Upcoming transactions do not appear in the account balance summary until its set date, they are visually distinct from the current ones and it's super useful.

> I made this exact request and was told that's what schedules are for.

Schedules are useful for the recurring transactions, but if it's a oneshot or in case the amount is variable creating them manually on the account is a **much more user-friendly** approach.

In the end, is this a question of product design or insufficient resources to implement this feature?

**Comment by alex-t-grey** [comment]
@d2718nis the 25.12.0 patch notes say that #6065 has been implemented already

**Comment by SLahti** [comment]
> [@d2718nis](https://github.com/d2718nis) the 25.12.0 patch notes say that [#6065](https://github.com/actualbudget/actual/pull/6065) has been implemented already

For me this works as a solution. However it still clogs the schedule page with "unnecessary" one offs until they are solved. It is good enough though.

**Comment by d2718nis** [comment]
> [@d2718nis](https://github.com/d2718nis) the 25.12.0 patch notes say that [#6065](https://github.com/actualbudget/actual/pull/6065) has been implemented already

Cool, thanks for letting me know! A bit weird that you need that extra step but it's a design choice and it's fine.

> For me this works as a solution. However it still clogs the schedule page with "unnecessary" one offs until they are solved. It is good enough though.

Maybe adding a quick filter for oneshot vs recurring schedules on that Schedules view would be a good idea?

**Comment by Crosis47** [comment]
> Maybe adding a quick filter for oneshot vs recurring schedules on that Schedules view would be a good idea?

I second this. I'd also like to see the one-off schedules self-delete upon entry into the register.
